monitoring/monitoring/utils.py

```diff
@@ -0,0 +1,218 @@
+from typing import Any
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+
+from database.connection import GreenDB
+from monitoring import CONNECTION_FOR_TABLE
+
+green_db = GreenDB()
+
+
+def dates() -> tuple:
+    """
+    Gets latest timestamp for green-db and sustainability-labels tables and converts it to date
+    format.
+
+    Returns:
+        List: containing date of last extraction from green-db and date of last update from
+        sustainability-labels
+    """
+    last_extraction = green_db.get_latest_timestamp().date()
+    last_label_update = green_db.last_update_sustainability_labels().date()
+    return last_extraction, last_label_update
+
+
+def all_scraping_summary() -> Any:
+    """
+    Uses 'get_scraping_summary' method to query all ScrapingDB tables.
+
+    Returns:
+        Dataframe: with number of products by country and merchant for all timestamps for all
+        existing tables in ScrapingDB
+    """
+    all_scraping = []
+    for value in CONNECTION_FOR_TABLE.values():
+        all_scraping.extend(value.get_scraping_summary())
+    return pd.DataFrame(
+        all_scraping, columns=["merchant", "timestamp", "products", "country"]
+    ).sort_values(by="merchant")
+
+
+def last_scraping_summary() -> Any:
+    """
+    Filters returned dataframe from 'all_scraping_summary' by latest timestamp found in green-db
+    table.
+
+    Returns:
+        Dataframe: with number of products by country and merchant for latest timestamp for all
+        existing tables in ScrapingDB
+    """
+    return all_scraping_summary()[
+        (all_scraping_summary()["timestamp"] >= green_db.get_latest_timestamp())
+    ]
+
+
+def all_extraction_summary() -> Any:
+    """
+    Uses 'get_extraction_summary' output and converts it in a dataframe.
+
+    Returns:
+        Dataframe: with number of products by country and merchant for all timestamps in green-db
+        table.
+    """
+    return pd.DataFrame(
+        green_db.get_extraction_summary(), columns=["merchant", "timestamp", "products", "country"]
+    ).sort_values(by="merchant")
+
+
+def last_extraction_summary() -> Any:
+    """
+    Filters returned dataframe from 'all_extraction_summary' by latest timestamp found in
+    green-db table.
+
+    Returns:
+        Dataframe: with number of products by country and merchant for latest timestamps in green-db
+        table.
+    """
+    return all_extraction_summary()[
+        (all_extraction_summary()["timestamp"] >= green_db.get_latest_timestamp())
+    ]
+
+
+def products_by_category() -> tuple:
+    """
+    Gets 'get_category_summary' output, converts it in a dataframe and creates a bar chart for
+    products per category.
+
+    Returns:
+        List: [plotly chart, pandas dataframe]
+    """
+    df = pd.DataFrame(green_db.get_category_summary(), columns=["category", "merchant", "products"])
```

Comment:
Merge this with the API code in `connection.py` and directly return a DataFrame

Reply:
Merged in connection.py ([6800091](https://github.com/calgo-lab/green-db/pull/91/commits/68000913abc6ced9cff53a0be8aa7df59596e658))